src/expr/proof_step_buffer.h

```diff
@@ -0,0 +1,91 @@
+/*********************                                                        */
+/*! \file proof_step_buffer.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Proof step and proof step buffer utilities.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__EXPR__PROOF_STEP_BUFFER_H
+#define CVC4__EXPR__PROOF_STEP_BUFFER_H
+
+#include <vector>
+
+#include "expr/node.h"
+#include "expr/proof_checker.h"
+#include "expr/proof_rule.h"
+
+namespace CVC4 {
+
+/** Information for constructing a step in a CDProof */
+class ProofStep
+{
+ public:
+  ProofStep();
+  ProofStep(PfRule r,
+            const std::vector<Node>& children,
+            const std::vector<Node>& args);
+  /** The proof rule */
+  PfRule d_rule;
+  /** The proof children */
+  std::vector<Node> d_children;
+  /** The proof arguments */
+  std::vector<Node> d_args;
+};
```

Comment:
Why isn't the conclusion of the step part of the proof step? In the usage below it seems everything would just be simpler if this was the case.

Reply:
I'd like to separate premises from conclusion, a proof step is everything that is necessary for deriving a conclusion.

Reply:
In particular, one reason for this separation: I am often mapping Node (conclusions) to their corresponding ProofStep.  Having conclusion within the ProofStep in this case is redundant.

Reply:
OK, that sounds good. Could you just add this motivation for the doc of the step? Might be good to have this motivation said explicitly.